ocaml/typing/typedecl.ml

```diff
@@ -1094,6 +1098,54 @@ let check_constraints env sdecl (_, decl) =
       check_constraints_rec env sty.ptyp_loc visited ty
   end
 
+(* CR layouts v3: this is a hack to allow re-exporting the definition
```

Comment:
I think this isn't quite v3, because we don't really have a plan for allowing user-written null constructors. Maybe make a v3.5 for this? I'm not sure we'll ever do it -- would need to study the level of need.